### ISSUE TITLE
"Mailto" links not detected correctly using `getUrls()`

### ISSUE THREAD

**Issue by bastones** [vstelmakh/url-highlight#48]
Please see example below.

**Code:**

```
(new UrlHighlight)->getUrls('<a href="mailto:hello@example.com">Example</a>')
```

**Actual output:**

```
[
   "href="mailto:hello@example.com",
]
```

**Expected output:**

```
[
   "hello@example.com",
]

// Or

[
   "mailto:hello@example.com",
]

// Or

[
   // An empty array
]
```

**Comment by vstelmakh** [comment]
Hi @bastones 👋,

Good catch, this is definitely not expected (correct) behaviour. Thank you for reporting the valid issue.

I had a quick look, and it seems like the regex in [Matcher](https://github.com/vstelmakh/url-highlight/blob/bcdaca1e22a69b85f0ee8ce534262461c0f4114c/src/Matcher/Matcher.php#L72) class is not properly matching the userinfo for this case. 

Unfortunately I'm a bit short on time at the moment. So I can't tackle it directly. But I believe it can be fixed.

**Comment by vstelmakh** [comment]
Adding a few notes and references for future myself.

userinfo = *( unreserved / pct-encoded / sub-delims / ":" )

* `unreserved` = ALPHA / DIGIT / "-" / "." / "_" / "~"
* `pct-encoded` (percent encoded) = "%" followed by the hexadecimal representation of the ASCII value
* `sub-delims` = "!" / "$" / "&" / "'" / "(" / ")" / "*" / "+" / "," / ";" / "="

References:
* [RFC 3986 Uniform Resource Identifier (URI): Generic Syntax](https://datatracker.ietf.org/doc/html/rfc3986)
* [RFC 6068  The 'mailto' URI Scheme](https://datatracker.ietf.org/doc/html/rfc6068)

**Comment by vstelmakh** [comment]
Issue is resolved in #49, and it's available from [v3.1.0](https://github.com/vstelmakh/url-highlight/releases/tag/v3.1.0).
